### ISSUE TITLE
Int64 column gets converted to in32?

### ISSUE THREAD

**Issue by rickbeeloo** [biocore-ntnu/pyranges#164]
When I read the `Start` column which is `int64` when I check it in Pandas it somehow appears to be `int23` after passing it to `PyRanges`, which I suppose is the cause of the following error when joining later on:

> Buffer dtype mismatch, expected 'const int64_t' but got 'int'

The code I use:
```
df = pd.read_csv(path, sep='\t', nrows = 1000)
print(df['Start'].dtype) #gives int64
anno_pr = pr.PyRanges(df)
print(anno_pr.head())

+------------------------------+--------------+-----------+-------+
| Chromosome                   | Strand       |     Start | +4    |
| (category)                   | (category)   |   (int32) | ...   |
|------------------------------+--------------+-----------+-------|
| 1733.8098_1733.8098.con.0001 | +            |        79 | ...   |
| 1733.8098_1733.8098.con.0001 | +            |       166 | ...   |
| 1733.8098_1733.8098.con.0001 | +            |      4820 | ...   |
| 1733.8098_1733.8098.con.0001 | +            |      7029 | ...   |
| 1733.8098_1733.8098.con.0001 | +            |      7978 | ...   |
| 1733.8098_1733.8098.con.0001 | +            |      9548 | ...   |
| 1733.8098_1733.8098.con.0001 | +            |     10002 | ...   |
| 1733.8098_1733.8098.con.0001 | +            |     10499 | ...   |
+------------------------------+--------------+-----------+-------+

```
Is there anything I should change? Maybe I missed something

**Comment by rickbeeloo** [comment]
Should we in this case always specifcy `int64=True`?

**Comment by endrebak** [comment]
Yeah, that is the way to do it. If you think of a backwards-compatible way to do it I'll listen :)

**Comment by rickbeeloo** [comment]
Hm no idea, but it now is kinda confusing that the type can change from pandas to pyranges. I would expect it to copy the `dtype` from Pandas when initiating the Pyranges object if it's not explicitly specified using e.g. `int64=True`. 

A totally different question, would it be possible to show the progress for a `join`? We are joining tables with over a million number of rows and around 400K chromosome IDs so it's hard to get a grasp of how long this would take without a progress estimate beforehand /  progressbar (I could make a seperate "issue" for this if that would help)

**Comment by endrebak** [comment]
For the latter you could use pyranges.itergrs:
https://pyranges.readthedocs.io/en/latest/autoapi/pyranges/#pyranges.itergrs

Then you’d iterate over the same chromosomes in each pyranges. Then you
could join per chromosome. Use
https://click.palletsprojects.com/en/7.x/utils/ to easily create a progress
bar. If you have further questions, feel free to create an issue :)

Endre
